OneSignalSDK/unittest/src/test/java/com/test/onesignal/InAppMessageIntegrationTests.java

```diff
@@ -633,6 +635,160 @@ public void testInAppMessageClickActionOutcomeWithValue() throws Exception {
+            add(message);
+        }});
+
+        // 1. Init OneSignal
+        OneSignalInit();
+        threadAndTaskWait();
+
+        // Enable influence outcomes
+        trackerFactory.saveInfluenceParams(new OneSignalPackagePrivateHelper.RemoteOutcomeParams());
+
+        final OSInAppMessageAction[] lastAction = new OSInAppMessageAction[1];
+        OneSignal.getCurrentOrNewInitBuilder().setInAppMessageClickHandler(new OneSignal.InAppMessageClickHandler() {
+            @Override
+            public void inAppMessageClicked(OSInAppMessageAction result) {
+                lastAction[0] = result;
+                // Ensure we are on the main thread when running the callback, since the app developer
+                //   will most likely need to update UI.
+                assertMainThread();
+
+                OneSignal.sendOutcome("test");
+                // Ensure outcome is sent
+                ShadowOneSignalRestClient.Request iamOutcomeRequest = ShadowOneSignalRestClient.requests.get(4);
+
+                assertEquals("outcomes/measure_sources", iamOutcomeRequest.url);
+                // Requests: Param request + Players Request + IAM Request + Impression + Outcome Request
+                assertEquals(5, ShadowOneSignalRestClient.requests.size());
+                assertFalse(iamOutcomeRequest.payload.has("weight"));
+                try {
+                    // Check outcome was DIRECT from IAM
+                    assertEquals("test", iamOutcomeRequest.payload.get("id"));
+                    assertEquals(1, iamOutcomeRequest.payload.get("device_type"));
+                    JSONObject sources = iamOutcomeRequest.payload.getJSONObject("sources");
+                    JSONObject direct = sources.getJSONObject("direct");
+                    JSONArray iamIds = direct.getJSONArray("in_app_message_ids");
+                    assertEquals(0, direct.getJSONArray("notification_ids").length());
+                    assertEquals(1, iamIds.length());
+                    assertEquals(message.messageId, iamIds.get(0));
+                    assertFalse(sources.has("indirect"));
+                } catch (JSONException e) {
+                    e.printStackTrace();
+                }
+            }
+        });
+        threadAndTaskWait();
+
+        // Add trigger to make IAM display
+        OneSignal.addTrigger("test_1", 2);
+        assertEquals(1, OneSignalPackagePrivateHelper.getInAppMessageDisplayQueue().size());
+
+        OneSignalPackagePrivateHelper.onMessageActionOccurredOnMessage(message,
+                new JSONObject() {{
+                    put("id", "button_id_123");
+                    put("name", "my_click_name");
+                }}
+        );
+
+        // Ensure we fire public callback that In-App was clicked.
+        assertEquals(lastAction[0].clickName, "my_click_name");
+    }
+
+    @Test
+    public void testOnMessageActionOccurredOnMessageSendOutcomeV2AfterDismiss() throws Exception {
```

Comment:
rename this test maybe?
`testIAMActionSendsOutcome_afterDismiss_usingOutcomesV2` or
`testOnIAMActionSendsOutcome_afterDismiss_usingOutcomesV2`